_posts/2022-09-20-Project-Sabine-Low-cost-DIY-underwater-modem-using-COTS-components-and-Unet-audio.md

```diff
@@ -0,0 +1,326 @@
+
+Next, wrap some cling wrap to waterproof the front portion. You can pour some hot glue on the outside surface of the bush and then wrap the cling wrap over the front of the microphone making sure that there is no gap between the cling wrap and the black diaphragm. If there is an air gap, then the performance might be affected.
+
+<p align="center"><img width="50%" src="../assets/img/sabine/hydrophone1.png"/></p>
+<p align="center"><img width="50%" src="../assets/img/sabine/hydrophone2.png"/></p>
+<p align="center"><em>Waterproofed mic</em></p>
+
+
+
+
+### Bill of materials
+
+| Item                                    | Qty  | Cost (USD) |
+| --------------------------------------- | ---- | ---------- |
+| Raspberry Pi                            | 1    | 129.00     |
+| USB sound card for RPi                  | 1    | 0.87       |
+| Power amplifier                         | 1    | 0.45       |
+| Underwater speaker                      | 1    | 95.50      |
+| Electret microphone                     | 1    | 0.99       |
+| 3.5mm audio jack                        | 2    | 1.36       |
+| Bush, glue gun, glue stick, cling wrap, | 1    | < 20       |
+| __Total__                               |      | 292.72     |
+
+> NOTE: It is recommended to get more than 1 for the electret microphone, audio jacks, etc. in case of damage during assembly
+
+## Software installation
+
+### Requirements
+
+- Raspberry Pi 3+
+- USB Sound Card
+- Headphones/Loudspeaker
+- Microphone
+
+### Setup
+
+#### OS 
+
+1. Install the latest [Raspbian](https://www.raspberrypi.org/software/operating-systems/) on your Raspberry Pi
+
+#### Audio
+
+1. Plug the USB audio card into your Raspberry Pi
+2. Power on your Raspberry Pi
+3. Ensure that the USB Audio Card is detected using `aplay -l`
+
+```
+> aplay -l
+card 0: ...
+card 1: Audio [USB Audio], device 0: USB Audio [USB Audio]
+  Subdevices: 0/1
+  Subdevice #0: subdevice #0
+```
+4. Disable the RaspberyPi onboard sound card, by commenting out the line `dtparam=audio=on` in `/boot/config.txt`
+
+```
+> sudo nano /boot/config.txt
+
+# Enable audio (loads snd_bcm2835)
+# dtparam=audio=on
+```
+
+5. Make the USB Audio card the default device in alsa by updating `defaults.ctl.card` and `defaults.pcm.card` to `1` in `/usr/share/alsa/alsa.conf`
+
+```
+defaults.ctl.card 1
+defaults.pcm.card 1
+```
+
+6. Reboot your Raspberry Pi and verify that `card 0:` is not enumerated by `aplay -l`
+
+```
+> aplay -l
+card 1: Audio [USB Audio], device 0: USB Audio [USB Audio]
+  Subdevices: 0/1
+  Subdevice #0: subdevice #0
+```
+
+7. You can test if your audio setup works properly using  `speaker-test -c2`
+
+### Dependencies
+
+1. Install Java8
+
+```sh
+wget https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/download/jdk8u265-b01/OpenJDK8U-jdk_arm_linux_hotspot_8u265b01.tar.gz
+tar xzf OpenJDK8U-jdk_arm_linux_hotspot_8u265b01.tar.gz
+rm jdk8u265-b01/src.zip
+sudo mkdir -p /usr/lib/jvm/java-8-openjdk-armhf
```

Comment:
Why are we using `armhf`? That's for 32bit ARM. I am guessing it will work because of backward compatibility, but maybe easier to use `arm64` or `aarch64` for longevity reasons.

Reply:
I copied these instructions from the document you provided. Perhaps we can test this with the next release in rpi and update later? These are the instructions that worked in the current setup.